### PR TITLE
Protocol Update.

### PR DESCRIPTION
**Pull request by greatman**
Updated ServerListPingCodec/Message/Event, EffectCodec/Message, PlayerLocaleViewDistanceCodec/Message, PlayerTimeCodec/Message

Didin't implement those stuff internally so it probably needs some tweaking like world age.

Signed-off-by: William Lebel lebelwilliam@gmail.com

### CODE REVIEW

File: src/main/java/org/spout/vanilla/VanillaPlugin.java [hunk 1]
```diff
@@ -93,7 +93,8 @@
 
 public class VanillaPlugin extends CommonPlugin {
 	private static final int LOADER_THREAD_COUNT = 16;
-	public static final int MINECRAFT_PROTOCOL_ID = 45;
+	public static final int MINECRAFT_PROTOCOL_ID = 46;
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/VanillaPlugin.java]: The latest protocol version is 47

File: src/main/java/org/spout/vanilla/protocol/msg/player/PlayerLocaleViewDistanceMessage.java [hunk 2]
```diff
@@ -101,4 +103,12 @@ public byte getDifficulty() {
 	public void setDifficulty(byte difficulty) {
 		this.difficulty = difficulty;
 	}
+
+	public boolean getShowCape() {
+		return showCape;
+	}
+
+	public void setShowCape(boolean showCape) {
```
Comment (zml2008) [inline src/main/java/org/spout/vanilla/protocol/msg/player/PlayerLocaleViewDistanceMessage.java]: Messages are immutable.

File: src/main/java/org/spout/vanilla/VanillaPlugin.java (hunk 1 above)
Comment (greatman) [inline src/main/java/org/spout/vanilla/VanillaPlugin.java]: Looks like http://www.wiki.vg/Pre-release_protocol is not updated then. Says 46 over there

Comment (zml2008) [inline src/main/java/org/spout/vanilla/VanillaPlugin.java]: use b.wiki.vg

Comment (greatman) [inline src/main/java/org/spout/vanilla/VanillaPlugin.java]: Oh forgot about that thing. Thanks :)

**Comment by greatman** [comment]
Updated.

File: src/main/java/org/spout/vanilla/protocol/msg/player/PlayerTimeMessage.java [hunk 3]
```diff
@@ -32,16 +32,21 @@
...
+	public PlayerTimeMessage(long time, long dayTime) {
 		this.time = time;
+		this.dayTime = dayTime;
 	}
 
 	public long getTime() {
 		return time;
 	}
 
+	public long getDayTime() {
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/protocol/msg/player/PlayerTimeMessage.java]: This is supposed to be the age of the world IIRC

File: src/main/java/org/spout/vanilla/event/game/ServerListPingEvent.java [hunk 4]
```diff
@@ -108,9 +110,41 @@ public void setMaxPlayers(int maxPlayers) {
 	 * @return packet message
 	 */
 	public String getMessage() {
-		return motd + "\u00A7" + numPlayers + "\u00A7" + maxPlayers + "\u00A7";
+		return "§1" + "\u0000" + protocolVersion + "\u0000" + protocolVersionString + "\u0000" + motd + "\u0000" + numPlayers + "\u0000" + maxPlayers + "\u0000";
 	}
 
+	/**
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/event/game/ServerListPingEvent.java]: What exactly is the purpose of the following methods?

File: src/main/java/org/spout/vanilla/VanillaPlugin.java [hunk 5]
```diff
@@ -93,7 +93,8 @@
 
 public class VanillaPlugin extends CommonPlugin {
 	private static final int LOADER_THREAD_COUNT = 16;
-	public static final int MINECRAFT_PROTOCOL_ID = 45;
+	public static final int MINECRAFT_PROTOCOL_ID = 47;
+	public static final String MINECRAFT_PROTOCOL_STRING = "1.3";
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/VanillaPlugin.java]: Why do we need this?

File: src/main/java/org/spout/vanilla/event/game/ServerListPingEvent.java (hunk 4 above)
Comment (greatman) [inline src/main/java/org/spout/vanilla/event/game/ServerListPingEvent.java]: Needed for the new layout of the packet. It's listed like that now: §1\047\012w42b\0Wiki.vg Minecraft Server\05\010 (protocol version 47, minecraft version 12w42b, motd 'Wiki.vg Minecraft Server', 5 users, 10 slots)

File: src/main/java/org/spout/vanilla/VanillaPlugin.java (hunk 5 above)
Comment (greatman) [inline src/main/java/org/spout/vanilla/VanillaPlugin.java]: As above.

File: src/main/java/org/spout/vanilla/event/game/ServerListPingEvent.java [hunk 6]
```diff
@@ -108,9 +110,41 @@ public void setMaxPlayers(int maxPlayers) {
 	 * @return packet message
 	 */
 	public String getMessage() {
-		return motd + "\u00A7" + numPlayers + "\u00A7" + maxPlayers + "\u00A7";
+		return "§1" + "\u0000" + protocolVersion + "\u0000" + protocolVersionString + "\u0000" + motd + "\u0000" + numPlayers + "\u0000" + maxPlayers + "\u0000";
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/event/game/ServerListPingEvent.java]: I'de rather you use `\u00A7` instead of `§`

Comment (greatman) [inline src/main/java/org/spout/vanilla/event/game/ServerListPingEvent.java]: Thanks for the tip.

**Comment by greatman** [comment]
Updated for Windwaker comments. Also fixed the version string to be 1.4 hehe.

**Comment by Zidane** [comment]
1 Commit please :p

**Pull request closed without merging**